charts/graphistry-helm/templates/telemetry/otel-collector.yaml

```diff
@@ -0,0 +1,137 @@
+{{ if eq .Values.telemetryEnv.ENABLE_OPEN_TELEMETRY true }}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    io.kompose.service: otel-collector
+  name: otel-collector
+spec:
+  selector:
+    matchLabels:
+      io.kompose.service: otel-collector
+  template:
+    metadata:
+      labels:
+        io.kompose.network/grph: "true"
+        io.kompose.service: otel-collector
+    spec:
+      {{ if eq .Values.telemetryEnv.OTEL_CLOUD_MODE false }}
+      initContainers:
+      - name: "{{ lower .Chart.Name }}-init-prometheus"
+        {{- if eq .Values.global.containerregistry.name "docker.io/graphistry" }}
+        image: "groundnuty/k8s-wait-for:latest" #DockerHub
+        {{ else }}
+        image: {{.Values.global.containerregistry.name}}/k8s-wait-for:latest
+        {{- end }}
+        resources:
+            {{- toYaml .Values.InitContainerResources | nindent 10 }}
+        imagePullPolicy: {{.Values.global.imagePullPolicy }} 
+        args: 
+        - "service"
+        - "prometheus"
+      - name: "{{ lower .Chart.Name }}-init-jaeger"
+        {{- if eq .Values.global.containerregistry.name "docker.io/graphistry" }}
+        image: "groundnuty/k8s-wait-for:latest" #DockerHub
+        {{ else }}
+        image: {{.Values.global.containerregistry.name}}/k8s-wait-for:latest
+        {{- end }}
+        resources:
+            {{- toYaml .Values.InitContainerResources | nindent 10 }}
+        imagePullPolicy: {{.Values.global.imagePullPolicy }} 
+        args: 
+        - "service"
+        - "jaeger"
+      {{- end }}
+      containers:
+        - env:
+            - name: NODE_ENV
+              value: {{ .Values.nodeEnv | quote }}
+          {{- range .Values.env }}
+            - name: {{ .name }}
+              value: {{ .value | quote }}
+          {{- end }}
+            - name: GRAPHISTRY_LOG_LEVEL
+              value: {{ .Values.global.logs.GraphistryLogLevel | quote }}
+            - name: LOG_LEVEL
+              value: {{ .Values.global.logs.LogLevel | quote }}
+            - name: OTEL_LOG_LEVEL
+              value: "warn"
+            - name: OTEL_EXPORTER_JAEGER_ENDPOINT
+              value: "jaeger:4317"
+          {{- if eq .Values.telemetryEnv.OTEL_CLOUD_MODE true }}
+            - name: OTEL_COLLECTOR_OTLP_USERNAME
+              value: {{ .Values.telemetryEnv.OTEL_COLLECTOR_OTLP_USERNAME | quote }}
+            - name: OTEL_COLLECTOR_OTLP_PASSWORD
+              value: {{ .Values.telemetryEnv.OTEL_COLLECTOR_OTLP_PASSWORD | quote }}
+            - name: OTEL_COLLECTOR_OTLP_HTTP_ENDPOINT
+              value: {{ .Values.telemetryEnv.OTEL_COLLECTOR_OTLP_HTTP_ENDPOINT | quote }}
+          {{- end }}
+          image: otel/opentelemetry-collector-contrib:0.87.0
```

Comment:
add helm templating to image and tag and control in values.yanl? makes updates easier?